python/lsst/ctrl/bps/wms/htcondor/htcondor_service.py

```diff
@@ -99,32 +106,22 @@ def report(self, wms_workflow_id=None, user=None, hist=0, pass_thru=None):
         """
         message = ""
 
-        if wms_workflow_id and isinstance(wms_workflow_id, str) and not wms_workflow_id.isnumeric():
-            # assume wms_workflow_id is actually the path
-            # Try to read from logs
-            jobs = read_node_status(wms_workflow_id)
-            jobs.update(read_dag_node_log(wms_workflow_id))
-        else:
-            if wms_workflow_id:
-                constraint = f'(DAGManJobId == {int(wms_workflow_id)} || ClusterId == ' \
-                             f'{int(wms_workflow_id)})'
-            elif pass_thru:
-                constraint = pass_thru
-            else:
-                constraint = 'bps_isjob == "True"'
-                if user:
-                    constraint += f' && (Owner == "{user}" || bps_operator == "{user}")'
-            # check runs in queue
-            jobs = condor_q(constraint)
+        if wms_workflow_id:
+            try:
+                float(wms_workflow_id)
+                is_float = True
+            except ValueError:  # Don't need TypeError here as None goes to else branch.
+                is_float = False
 
-            if hist:
-                epoch = (datetime.now() - timedelta(days=hist)).timestamp()
-                constraint += f' && (CompletionDate >= {epoch} || JobFinishedHookDone >= {epoch})'
-                hist_jobs = condor_history(constraint)
-                jobs.update(hist_jobs)
+            if is_float:
+                run_reports, message = _report_from_id(float(wms_workflow_id), hist)
+            else:
+                run_reports, message = _report_from_path(wms_workflow_id)
+        else:
+            run_reports, message = summary_report(user, hist, pass_thru)
+        _LOG.debug("report: %s, %s", run_reports, message)
```

Comment:
Is there a reason why you couldn't use `isinstance()` instead of `try-except`?  Unless I'm missing something, lines 109-122 could be replaced with:

```python
if wms_workflow_id is not None:
    if isinstance(wms_workflow_id, float):
        ...
    else:
        ...
else:
    ...
```

and it would work equally well in this case, but would be more straightforward.